miq_version/__init__.py

```diff
@@ -402,11 +422,20 @@ def parse_template(cls, template_name):
                         template_date = futurecheck(date(year, month, day))
                     except ValueError:
                         continue
-                    dot_version = (version if 'downstream' not in stream_tuple.stream else
-                                   '{}.{}.{}.{}'.format(version[0],
-                                                        version[1],
-                                                        version[2],
-                                                        version[3:]))
+                    if 'downstream' not in stream_tuple.stream:
+                        dot_version = version
+                    elif version.startswith('51'):
+                        version = version.ljust(4, '0')
```

Comment:
What's the goal here with an `ljust`?  Please comment in-line, wondering if your intent is to do some zero-padding, which we need to be careful about or are better off not doing unless there's ambiguity.

Reply:
there are templates which have only 2 numbers like s_tpl_downstream_58z_100403_blabla.
I can add some parsing which would suit any amound of version chars. However, ljust is the easiest way to make such versions suit existing pattern.